src/server/services/wms/qgswmsgetcapabilities.cpp

```diff
@@ -703,6 +759,945 @@ namespace QgsWms
+        {
+          appendCrsElementToLayer( doc, layerElement, CRSPrecedingElement, crs );
+        }
+      }
+
+      //Support for CRS:84 is mandatory (equals EPSG:4326 with reversed axis)
+      appendCrsElementToLayer( doc, layerElement, CRSPrecedingElement, QString( "CRS:84" ) );
+    }
+
+    void appendCrsElementToLayer( QDomDocument &doc, QDomElement &layerElement, const QDomElement &precedingElement,
+                                  const QString &crsText )
+    {
+      if ( crsText.isEmpty() )
+        return;
+      QString version = doc.documentElement().attribute( QStringLiteral( "version" ) );
+      QDomElement crsElement = doc.createElement( version == QLatin1String( "1.1.1" ) ? "SRS" : "CRS" );
+      QDomText crsTextNode = doc.createTextNode( crsText );
+      crsElement.appendChild( crsTextNode );
+      layerElement.insertAfter( crsElement, precedingElement );
+    }
+
+    void appendLayerBoundingBoxes( QDomDocument &doc, QDomElement &layerElem, const QgsRectangle &lExtent,
+                                   const QgsCoordinateReferenceSystem &layerCRS, const QStringList &crsList,
+                                   const QStringList &constrainedCrsList )
+    {
+      if ( layerElem.isNull() )
+      {
+        return;
+      }
+
+      QgsRectangle layerExtent = lExtent;
+      if ( qgsDoubleNear( layerExtent.xMinimum(), layerExtent.xMaximum() ) || qgsDoubleNear( layerExtent.yMinimum(), layerExtent.yMaximum() ) )
+      {
+        //layer bbox cannot be empty
+        layerExtent.grow( 0.000001 );
+      }
+
+      QgsCoordinateReferenceSystem wgs84 = QgsCoordinateReferenceSystem::fromOgcWmsCrs( GEO_EPSG_CRS_AUTHID );
+
+      QString version = doc.documentElement().attribute( QStringLiteral( "version" ) );
+
+      //Ex_GeographicBoundingBox
+      QDomElement ExGeoBBoxElement;
+      //transform the layers native CRS into WGS84
+      QgsRectangle wgs84BoundingRect;
+      if ( !layerExtent.isNull() )
+      {
+        QgsCoordinateTransform exGeoTransform( layerCRS, wgs84 );
+        try
+        {
+          wgs84BoundingRect = exGeoTransform.transformBoundingBox( layerExtent );
+        }
+        catch ( const QgsCsException & )
+        {
+          wgs84BoundingRect = QgsRectangle();
+        }
+      }
+
+      if ( version == QLatin1String( "1.1.1" ) ) // WMS Version 1.1.1
+      {
+        ExGeoBBoxElement = doc.createElement( QStringLiteral( "LatLonBoundingBox" ) );
+        ExGeoBBoxElement.setAttribute( QStringLiteral( "minx" ), QString::number( wgs84BoundingRect.xMinimum() ) );
+        ExGeoBBoxElement.setAttribute( QStringLiteral( "maxx" ), QString::number( wgs84BoundingRect.xMaximum() ) );
+        ExGeoBBoxElement.setAttribute( QStringLiteral( "miny" ), QString::number( wgs84BoundingRect.yMinimum() ) );
+        ExGeoBBoxElement.setAttribute( QStringLiteral( "maxy" ), QString::number( wgs84BoundingRect.yMaximum() ) );
+      }
+      else // WMS Version 1.3.0
```

Comment:
With the new refactoring, as @dmarteau explained in Lyon, there is support for services handling different versions of the same protocol, so, why don't move this 1.3.0 code into its own service module? It could be a subclass of the1.1.1 module that only re-defines what needs to be changed.
I think that this would make it easier to maintain and extend the code to handle further versions (if any) and if we need to apply a patch for a particular version we immediately know where to look.